dpnp/linalg/dpnp_iface_linalg.py

```diff
@@ -1560,3 +2025,195 @@ def tensorsolve(a, b, axes=None):
+    >>> np.linalg.trace(a).shape
+    (3, 2)
+
+    Traces adjacent to the main diagonal can be obtained by using the
+    `offset` argument:
+
+    >>> a = np.arange(9).reshape((3, 3)); a
+    array([[0, 1, 2],
+           [3, 4, 5],
+           [6, 7, 8]])
+    >>> np.linalg.trace(a, offset=1)  # First superdiagonal
+    array(6)
+    >>> np.linalg.trace(a, offset=2)  # Second superdiagonal
+    array(2)
+    >>> np.linalg.trace(a, offset=-1)  # First subdiagonal
+    array(10)
+    >>> np.linalg.trace(a, offset=-2)  # Second subdiagonal
+    array(6)
+
+    """
+
+    return dpnp.trace(x, offset, axis1=-2, axis2=-1, dtype=dtype)
+
+
+def vector_norm(x, /, *, axis=None, keepdims=False, ord=2):
+    """
+    Computes the vector norm of a vector (or batch of vectors) `x`.
+
+    This function is Array API compatible.
+
+    For full documentation refer to :obj:`numpy.linalg.vector_norm`.
+
+    Parameters
+    ----------
+    x : {dpnp.ndarray, usm_ndarray}
+        Input array.
+    axis : {None, int, n-tuple of ints}, optional
+        If an integer, `axis` specifies the axis (dimension) along which
+        to compute vector norms. If an n-tuple, `axis` specifies the axes
+        (dimensions) along which to compute batched vector norms. If ``None``,
+        the vector norm must be computed over all array values (i.e.,
+        equivalent to computing the vector norm of a flattened array).
+        Default: ``None``.
+    keepdims : bool, optional
+        If this is set to ``True``, the axes which are normed over are left in
+        the result as dimensions with size one. With this option the result
+        will broadcast correctly against the original `x`.
+        Default: ``False``.
+    ord : {int, inf, -inf, 'fro', 'nuc'}, optional
+        The order of the norm. For details see the table under ``Notes``
+        section in :obj:`dpnp.linalg.norm`.
+        Default: ``2``.
+
+    Returns
+    -------
+    out : dpnp.ndarray
+        Norm of the vector.
+
+    See Also
+    --------
+    :obj:`dpnp.linalg.norm` : Generic norm function.
+
+    Examples
+    --------
+    >>> import dpnp as np
+    >>> a = np.arange(9) + 1
+    >>> a
+    array([1, 2, 3, 4, 5, 6, 7, 8, 9])
+    >>> b = a.reshape((3, 3))
+    >>> b
+    array([[1, 2, 3],
+           [4, 5, 6],
+           [7, 8, 9]])
+
+    >>> np.linalg.vector_norm(b)
+    array(16.88194302)
+    >>> np.linalg.vector_norm(b, ord=np.inf)
+    array(9.)
+    >>> np.linalg.vector_norm(b, ord=-np.inf)
+    array(1.)
+
+    >>> np.linalg.vector_norm(b, ord=1)
+    array(45.)
+    >>> np.linalg.vector_norm(b, ord=-1)
+    array(0.35348576)
+    >>> np.linalg.vector_norm(b, ord=2)
+    array(16.881943016134134)
+    >>> np.linalg.vector_norm(b, ord=-2)
+    array(0.8058837395885292)
+
+    """
+    dpnp.check_supported_arrays_type(x)
+    x_shape = list(x.shape)
+    if axis is None:
+        # Note: dpnp.linalg.norm() doesn't handle 0-D arrays
+        x = x.ravel()
```

Comment:
It might be a dpctl tensor
```suggestion
        x = dpnp.ravel(x)
```